docs/reference/migration/migrate_7_10.asciidoc

```diff
@@ -18,54 +18,133 @@ See also <<release-highlights>> and <<es-release-notes>>.
 //tag::notable-breaking-changes[]
 
 [discrete]
-[[breaking_710_search_changes]]
-=== Search Changes
+[[breaking_710_security_changes]]
+=== Authentication changes
 
-[discrete]
-==== Missing checks for `index.max_docvalue_fields_search`
-The setting `index.max_docvalue_fields_search` limits the number of fields that
-can be retrieved through `docvalue_fields` in a search. Previously we only
-checked this limit when `docvalue_fields` appeared in the top-level search
-request body. Now we also enforce the limit when loading doc value fields in
-an `inner_hits` section or in a `top_hits` aggregation.
+[[api-keys-require-name-property]]
+.API keys now require a `name` property.
+[%collapsible]
+====
+*Details* +
+The `name` property is now required to create or grant an API key.
 
-[discrete]
-[[breaking_710_networking_changes]]
-=== Networking Changes
+[source,js]
+----
+{
+    "...": "...",
+    "api_key": {
+      "name": "key-1"
+    }
+}
+----
+// NOTCONSOLE
+
+*Impact* +
+To avoid errors, specify the `name` property when creating or granting API keys.
+====
 
 [discrete]
-==== *.tcp.keep_idle and *.tcp.keep_interval now bounded
-The settings `{network,transport,http}.tcp.keep_idle` and
-`{network,transport,http}.tcp.keep_interval` are now limited to a maximum
-value of 300, which amounts to 5 minutes. This helps ensure that, if
-explicitly configured, only reasonable values are set.
+[[breaking_710_java_changes]]
+==== Java changes
 
-//end::notable-breaking-changes[]
+[discrete]
+==== Machine Learning: `allow_no_jobs` and `allow_no_datafeeds` deprecated in favor of `allow_no_match`
+`allow_no_jobs` and `allow_no_datafeeds` API parameters are deprecated in favor
+of `allow_no_match`. The old parameters are still accepted by the APIs but the
+deprecation warning is emitted if the old parameter name is used in the request
+body or as a request param.
+High-level REST client classes will now send the new `allow_no_match` parameter.
 
-.Repository Stats API is deprecated
+[[supplier-searchlookup-arg]]
+.The `MappedFieldType#fielddataBuilder` method now accepts a `Supplier<SearchLookup>` argument.
 [%collapsible]
 ====
 *Details* +
-The Repository Stats API has been introduced in 7.8.0 as an experimental API
-and was never released. This API is superseded by the <<repositories-metering-apis,Repositories Metering APIs>>
-added in 7.10.0 which should be used instead. The Repository Stats API is
-deprecated starting 7.10.0 and will be removed in 8.0.0.
-
+To support future feature development, the existing
+`MappedFieldType#fielddataBuilder` method now accepts a new
+`Supplier<SearchLookup>` argument.
+ 
 *Impact* +
-Use the <<repositories-metering-apis,Repositories Metering APIs>>.
+If developing or maintaining an {es} plugin, update your implementation of the
```

Comment:
maybe we can be more precise and say "if you implement a mapper plugin" as not all plugins need to implement that method. But maybe it's too much, up to you :)

Reply:
Thanks for the feedback! I updated to `If you develop or maintain a mapper plugin,...`

Reply:
thank you!